src/modules/Dropdown/Dropdown.js

```diff
@@ -1167,6 +1169,7 @@ export default class Dropdown extends Component {
       onClick: this.handleItemClick,
       selected: selectedIndex === i,
       ...opt,
+      key: getKeyOrValue(opt.key, opt.value),
```

Comment:
Yes, I completely agree :+1:  Let's ship this fix as `0.68.5` and remove this line in `0.69.0`, and mark PR that makes this change as `breaking`.